### PR TITLE
Move P2P, RPC, historical sync to new BlobSidecar

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
Still polishing.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by zilm13** [comment]
Ready for review
- Removed test `missedBlobSidecarIndex` in BlobSidecarsByRangeListenerValidatingProxyTest is the same as `blobSidecarIndexIsInTheSameBlockButNotNext`
- Integration RPC tests cannot be transferred to the new BlobSidecar before storage changes

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxy.java [hunk 1]
```diff
@@ -84,55 +84,49 @@ private boolean blobSidecarSlotIsInRange(final UInt64 blobSidecarSlot) {
...
-          }
-        },
-        () -> {
-          // first blobSidecar
-          if (!blobSidecarSummary.index.isZero()) {
-            throw new BlobSidecarsResponseInvalidResponseException(
-                peer, BLOB_SIDECAR_UNEXPECTED_INDEX);
-          }
-        });
+    // It's first blobSidecar in response
```
Comment (tbenr) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxy.java:87]: 🤣

File: networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxyTest.java [hunk 2]
```diff
@@ -214,35 +206,23 @@ void blobSidecarIndexIsGreaterOrEqualThanMaxBlobs() {
...
-    safeJoin(listenerWrapper.onResponse(blobSidecar1));
-    safeJoin(listenerWrapper.onResponse(blobSidecar2));
-    safeJoin(listenerWrapper.onResponse(blobSidecar3));
-    safeJoin(listenerWrapper.onResponse(blobSidecar4));
-    safeJoin(listenerWrapper.onResponse(blobSidecar5));
-    safeJoin(listenerWrapper.onResponse(blobSidecar6));
-
-    final SafeFuture<?> result = listenerWrapper.onResponse(blobSidecar7);
+    final SignedBeaconBlock block1 = dataStructureUtil.randomSignedBeaconBlock(ONE);
+    final BlobSidecar blobSidecar10 = dataStructureUtil.randomBlobSidecarForBlock(block1, 0);
```
Comment (tbenr) [inline networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxyTest.java]: My brain suffers with this numbering. what about `blobSidecar1_0` or `blobSidecar1i0` or something else. Here and the other tests too.

File: networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxyTest.java [hunk 3]
```diff
@@ -361,38 +331,6 @@ void firstBlobSidecarIndexInNextBlockIsNotZero() {
                 .describe());
   }
 
-  @Test
```
Comment (tbenr) [inline networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxyTest.java:364]: why `missedBlobSidecarIndex` is gone?

**Review by tbenr — APPROVED** [review]
LGTM. just a couple of things.

File: networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxyTest.java (hunk 3 above)
Comment (zilm13) [reply networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxyTest.java:364]: Removed test missedBlobSidecarIndex in BlobSidecarsByRangeListenerValidatingProxyTest is the same as blobSidecarIndexIsInTheSameBlockButNotNext

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxy.java (hunk 1 above)
Comment (zilm13) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeListenerValidatingProxy.java:87]: 😇

**Pull request merged**